File: backend/src/main/java/com/musicmatch/backend/dto/PublicProfileResponse.java
package com.musicmatch.backend.dto;

import java.util.List;

import com.musicmatch.backend.model.ExperienceLevel;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
public class PublicProfileResponse {
    private Long id;
    private String username;
    private String biography;
    private String cityName;
    private Long cityId;
    private String profilePicture;
    private List<Long> styleIds;
    private List<InstrumentLevelResponse> instruments;
    private ExperienceLevel experienceLevel;

    private boolean blockedByMe;
    private boolean blockedMe;

    private String chatStatus;
    private Long chatId;
}

File: mobile/app/src/main/java/com/musicmatch/mobile/model/dto/PublicProfileResponse.kt
package com.musicmatch.mobile.model.dto

import com.musicmatch.mobile.model.ExperienceLevel

data class PublicProfileResponse(
    val id: Long,
    val username: String,
    val biography: String?,
    val cityName: String?,
    val cityId: Long?,
    val profilePicture: String?,
    val styleIds: List<Long>,
    val instruments: List<InstrumentLevelResponse>,
    val experienceLevel: ExperienceLevel,

    val blockedByMe: Boolean,
    val blockedMe: Boolean,

    val chatStatus: String?,
    val chatId: Long?
)